server/server.js

```diff
@@ -126,7 +126,11 @@ UserPresence = {
 			connection.metadata = metadata;
 		}
 
-		UsersSessions.upsert(query, update);
+		var result = UsersSessions.upsert(query, update);
+
+		if (result && connection.closed) {
+            UserPresence.removeConnection(connection.id);
```

Comment:
will this remove the connection created right above?

Reply:
I made a mistake.